tritium-metrics-jvm/src/main/java/com/palantir/tritium/metrics/jvm/JvmMetrics.java

```diff
@@ -132,48 +135,75 @@ private static void registerJvmMemory(TaggedMetricRegistry registry) {
             log.warn("Failed to load the MemoryMXBean, jvm.memory metrics will not be recorded");
             return;
         }
+        registerJvmMemory(registry, memoryBean);
+    }
+
+    @VisibleForTesting
+    static void registerJvmMemory(TaggedMetricRegistry registry, MemoryMXBean memoryBean) {
         JvmMemoryMetrics metrics = JvmMemoryMetrics.of(registry);
         // jvm.memory.total
-        metrics.totalInit((Gauge<Long>) () -> memoryBean.getHeapMemoryUsage().getInit()
-                + memoryBean.getNonHeapMemoryUsage().getInit());
-        metrics.totalUsed((Gauge<Long>) () -> memoryBean.getHeapMemoryUsage().getUsed()
-                + memoryBean.getNonHeapMemoryUsage().getUsed());
-        metrics.totalMax((Gauge<Long>) () -> memoryBean.getHeapMemoryUsage().getMax()
-                + memoryBean.getNonHeapMemoryUsage().getMax());
-        metrics.totalCommitted(
-                (Gauge<Long>) () -> memoryBean.getHeapMemoryUsage().getCommitted()
-                        + memoryBean.getNonHeapMemoryUsage().getCommitted());
+        metrics.totalInit((Gauge<Long>) () -> {
+            long heap = memoryBean.getHeapMemoryUsage().getInit();
+            long nonHeap = memoryBean.getNonHeapMemoryUsage().getInit();
+            return (heap < 0 || nonHeap < 0) ? null : heap + nonHeap;
```

Comment:
I assume the intent here to avoid producing any value if one don't have just one component (e.g. if we have heap init but do not have non-heap init)?

Reply:
I don't have super strong feelings, but curious for thoughts on https://github.com/palantir/tritium/pull/1393

Looking at spec, it seems like [`getInit()`](https://docs.oracle.com/en/java/javase/11/docs/api/java.management/java/lang/management/MemoryUsage.html#getInit()) & [`getMax()`](https://docs.oracle.com/en/java/javase/11/docs/api/java.management/java/lang/management/MemoryUsage.html#getMax()) may return -1 when undefined, but used & committed should be non-negative though I guess we should handle similarly in case